modules/rest-util/test/blaze/middleware/fhir/output_test.clj

```diff
@@ -205,5 +220,37 @@
       [:headers "Content-Type"] := "application/fhir+xml;charset=utf-8"
       [:body parse-xml :issue 0 :diagnostics] := "Invalid white space character (0x1e) in text to output (in xml 1.1, could output as a character entity)")))
 
+(deftest binary-resource-test
+
+  (testing "with data and with content type"
+    (given (call (binary-resource-handler-200 {:content-type "text/plain" :data "MTA1NjE0Cg=="}) {:headers {"accept" "text/plain"}})
+      :status := 200
+      [:headers "Content-Type"] := "text/plain"
+      [:body bs/from-byte-array] := #blaze/byte-string"3130353631340A"))
+
+  (testing "with data and without content type"
+    (given (call (binary-resource-handler-200 {:content-type nil :data "MTA1NjE0Cg=="}) {:headers {"accept" "text/plain"}})
+      :status := 200
+      [:headers "Content-Type"] := "application/octet-stream"
+      [:body bs/from-byte-array] := #blaze/byte-string"3130353631340A"))
+
+  (testing "without data and with content type"
+    (given (call (binary-resource-handler-200 {:content-type "text/plain"}) {:headers {"accept" "text/plain"}})
+      :status := 200
+      [:headers "Content-Type"] := "text/plain"
+      :body := nil))
+
+  (testing "without data and without content type"
+    (given (call (binary-resource-handler-200 {:content-type nil}) {:headers {"accept" "text/plain"}})
+      :status := 200
+      [:headers "Content-Type"] := "application/octet-stream"
+      :body := nil))
+
+  (testing "without body at all"
```

Comment:
this test will create a body, because you always create a binary resource. I would use the `resource-handler-304` instead. it creates a 304 response without a body.